include/seqan3/alignment/pairwise/align_pairwise.hpp

```diff
@@ -197,13 +160,34 @@ constexpr auto align_pairwise(exec_policy_t const & exec,
                   "Alignment configuration error: The sequence must model random_access_range and sized_range.");
 
     // Pipe with views::persist to allow rvalue non-view ranges.
-    auto seq_view = std::forward<sequence_t>(seq) | views::persist;
+    auto seq_view = std::forward<sequence_t>(sequences) | views::persist;
     // Configure the alignment algorithm.
     auto kernel = detail::alignment_configurator::configure<decltype(seq_view)>(config);
+
     // Create a two-way executor for the alignment.
-    detail::alignment_executor_two_way executor{std::move(seq_view), kernel, exec};
-    // Return the range over the alignments.
-    return alignment_range{std::move(executor)};
+    if constexpr (std::remove_reference_t<alignment_config_t>::template exists<align_cfg::parallel>())
+    {
+        if constexpr (std::remove_reference_t<alignment_config_t>::template exists<detail::vectorise_tag>())
+        {
+            detail::alignment_executor_two_way executor{std::move(seq_view), kernel, seqan3::par_unseq};
+            return alignment_range{std::move(executor)};
+        }
+        else
+        {
+            detail::alignment_executor_two_way executor{std::move(seq_view), kernel, seqan3::par};
+            return alignment_range{std::move(executor)};
+        }
+    }
+    else if constexpr (std::remove_reference_t<alignment_config_t>::template exists<detail::vectorise_tag>())
+    {
+        detail::alignment_executor_two_way executor{std::move(seq_view), kernel, seqan3::unseq};
+        return alignment_range{std::move(executor)};
+    }
+    else
+    {
+        detail::alignment_executor_two_way executor{std::move(seq_view), kernel, seqan3::seq};
+        return alignment_range{std::move(executor)};
+    }
```

Comment:
Is there a better way to do this so that you don't have so many redundant lines?

Reply:
We need to translate existing tags (parallel / vectorise) into an execution rule (seq, par_unseq, unseq, par) and haven't found another way to do it.
We tried different ways and this was the shortest one. As we look at two different config parts and their existence we can't use 'switch'.
But I would be very happy if you have ideas how to shorten this.

Reply:
You could do (I post the whole function just to be clear):

```cpp
template <typename sequence_t, typename alignment_config_t>
    requires detail::align_pairwise_range_input_concept<sequence_t> &&
             detail::is_type_specialisation_of_v<alignment_config_t, configuration>
constexpr auto align_pairwise(sequence_t && sequences,
                              alignment_config_t const & config)
{
    using first_seq_t  = std::tuple_element_t<0, value_type_t<sequence_t>>;
    using second_seq_t = std::tuple_element_t<1, value_type_t<sequence_t>>;

    static_assert(std::ranges::random_access_range<first_seq_t> && std::ranges::sized_range<first_seq_t>,
                  "Alignment configuration error: The sequence must model random_access_range and sized_range.");
    static_assert(std::ranges::random_access_range<second_seq_t> && std::ranges::sized_range<second_seq_t>,
                  "Alignment configuration error: The sequence must model random_access_range and sized_range.");

    // Pipe with views::persist to allow rvalue non-view ranges.
    auto seq_view = std::forward<sequence_t>(sequences) | views::persist;
    // Configure the alignment algorithm.
    auto kernel = detail::alignment_configurator::configure<decltype(seq_view)>(config);

    constexpr auto get_config = [] ()
    {
        using config_t = std::remove_reference_t<alignment_config_t>;

        if constexpr (config_t::template exists<align_cfg::parallel>())
        {
            if constexpr (config_t::template exists<detail::vectorise_tag>())
            {
                return seqan3::par_unseq;
            }
            else
            {
                return seqan3::par;
            }
        }
        else if constexpr (config_t::template exists<detail::vectorise_tag>())
            return seqan3::unseq;
        else
            return seqan3::seq;
    };

    // Create a two-way executor for the alignment.
    detail::alignment_executor_two_way executor{std::move(seq_view), kernel, get_config()};
    return alignment_range{std::move(executor)};
}
```

Then the choice of the tag is delegated to the lambda and the construction of the executor and the return of the function is very clear.

Maybe you can think of a better name than `get_config` ... something like `parallel_tag` that also indicates vectorised... :)